pyaerocom/io/read_airnow.py

```diff
@@ -445,3 +559,139 @@ def read(self, vars_to_retrieve=None, first_file=None, last_file=None):
         )
 
         return data
+
+    # the fllowing has been stolen from
+    # https://stackoverflow.com/questions/44402983/how-to-read-the-file-without-encoding-and-extract-desired-urls-with-python3
+    def get_file_bom_encoding(self, filename):
+        with open(filename, "rb") as openfileobject:
+            line = str(openfileobject.readline())
+            if line[2:14] == str(codecs.BOM_UTF8).split("'")[1]:
+                return "utf_8"
+            if line[2:10] == str(codecs.BOM_UTF16_BE).split("'")[1]:
+                return "utf_16"
+            if line[2:10] == str(codecs.BOM_UTF16_LE).split("'")[1]:
+                return "utf_16"
+            if line[2:18] == str(codecs.BOM_UTF32_BE).split("'")[1]:
+                return "utf_32"
+            if line[2:18] == str(codecs.BOM_UTF32_LE).split("'")[1]:
+                return "utf_32"
+        return ""
+
+    def get_all_file_encodings(self, filename):
+        encoding_list = []
+        encodings = (
```

Comment:
Maybe we could put this in a different file and read it in?

Reply:
It's really only used here. I don't see the point really.